frontend/app/components/redesign/components/ToolsWalletAddress.tsx

```diff
@@ -147,11 +150,12 @@ export const ToolsWalletAddress = () => {
         'flex flex-col xl:flex-row xl:items-start gap-2xl p-md bg-white rounded-lg',
         snap.walletConnectStep === 'error' && 'border border-red-600'
       )}
-      aria-labelledby={generatedId}
     >
       <div className="flex flex-col items-start gap-md w-full xl:flex-1 xl:grow">
         <div className="inline-flex items-center gap-xs">
-          <Heading5 id={generatedId}>Wallet address</Heading5>
+          <label htmlFor={generatedId}>
+            <Heading5 as="span">Wallet address</Heading5>
+          </label>
```

Comment:
Nit: Can we use `Heading5 as="label"`? Not sure if it's better or worse as code, but output wise, if we can add the style to label directly, that's less div-soup.